events/git-and-github.md

```diff
@@ -0,0 +1,126 @@
+- For Windows users, make that they install `git bash`
+- How to verify that git is installed
+- Instructions on how to create a Github Account
+
+In the post mention the fact that we will be available for installation issues before 15 minutes of the workshop. Help with the installation issues before the workshop begins.
+
+## Flow of the workshop
+
+0. **How will the workshop proceed**
+
+   Explain that Git and Github are important parts of Software Development. Tell them how the workshop will proceed like when should audience ask queries. Tell the audience that doing hands-on by themselves along with the workshop will help them learn better. Encourage them to ask questions, as some of the Git concepts are non-intuitive.
+
+1. **Terminal Commands**
+
+   It might be a good idea to introduce basic linux Commands, as they will be used through out the workshop. You don't have to cover this topic in detail. Just important commands like - `ls`, `cat`, `echo` , `cd`, `mkdir`, `pwd` , `rm`would suffice. Windows users can be asked to try these commands on Git Bash. 
+
+2. **Why do we need something like Git**
+
+   Before teaching anything, it is important that audience understand why they are learning it. Start with an example, with which they can connect. Note that most of the audience might have not have collaborated on a software. You can give an example of a folder, containing so many files, which are regularly updated. Tell them that tools called "Version Control Systems" are used to track projects. Tell them the importance of maintaining History and backups. Then you can tell that how such tools are useful in Software development and conclude that Git is used to track Software projects.
+
+3. **Mental Model of Git**:
+
+   Directly jumping to the terminal would confuse the audience. It might be frustrating for audience to type commands that they don't understand. Hence, first explain how Git tracks your project. Tell that by Git stores snapshots of the project. And each snapshot is referred as commit. You can show a `git log` of a good project, to convey the idea of how snapshots are taken at every feature. 
+
+   Explain the idea of staging area very carefully and clearly. Tell them that before storing snapshots, it is important to add them to a staging area. Show them a picture depicting the workflow - 1. git add 2. git commit.
+
+   <u>Note</u>: Though it might be tempting to introduce the concepts of Decentralized VCS and Centralized VCS, Merkle Tree. Please refrain from jumping to complex concepts. We have only 3 hours. It is important that we convey them knowledge of practical usage. One can use Git comfortably without knowing Merkle Trees. But making audience comfortable with Git requires some time and effort. So, its worth investing time in making the basics strong. However, you can share the links/resources to such concepts at the end of the workshop for the curious audience.
+
+4. **A demo**
```

Comment:
could we also tell them about .gitignore?

Reply:
In the beginning itself? It's a small thing, which can be discussed at the end. Will, it not unnecessarily cause confusion

Reply:
oh sure,better it is at the end.